scripts/check-ci-job-skippability.js

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env node
+
+/*
+
+This script takes two parameters or three environment variables:
+
+  {CI job name|$CIRCLE_JOB} {SHA1...SHA1|$CIRCLE_MERGE_BASE...$CIRCLE_SHA1}
+
+The flow of this script:
+
+  - List of files changed in PR: `git diff --name-only start...end`
+  - Extract list into array.
+  - Remove items from array if they match:
+    - Global ignore list.
+    - Other platform ignore lists.
+  - If array still contains items, do not authorize skipping CI job.
+  - Else if array is empty, skip CI job.
+
+*/
+
+const execSync = require('child_process').execSync;
+const _ = require('lodash');
+const minimatch = require("minimatch");
+
+console.step = _.partial(console.log, '\n\033[1m\033[36m*', _, '\033[0m');
+
+// BUILD PARAMETERS
+const CI_JOB_NAME = process.argv[2] || process.env.CIRCLE_JOB;
+if (!CI_JOB_NAME) { console.error('You must specify the name of a CI job as a parameter or environment variable.'); process.exit(1); }
+
+const COMMIT_RANGE = process.argv[3] || `${process.env.CIRCLE_MERGE_BASE}...${process.env.CIRCLE_SHA1}`;
+if (!COMMIT_RANGE || COMMIT_RANGE.includes('undefined')) { console.error('You must specify a range of commits to check for changed files in as a parameter or environment variable.'); process.exit(1); }
+
+// FILE SKIPPING PATTERNS
+const ALWAYS_SKIPPABLE_FILES = [
+  '**/CHANGELOG.md',
+  '**/README.md',
+  '**/CONTRIBUTING.md',
+  '**/DEVELOPING.md',
+  '**/INSTALL.md',
+  '**/*.podspec'
+]
+
+const ANDROID_FILES = [
+  'platform/android/**'
+]
+
+const IOS_FILES = [
+  'platform/ios/**',
+  'platform/darwin/**'
+]
+
+const MACOS_FILES = [
+  'platform/macos/**',
+  'platform/darwin/**'
+]
+
+const OTHER_PLATFORMS = [
+  'platform/glfw/**',
+  'platform/linux/**',
+  'platform/node/**',
```

Comment:
Two things: 1) this is my favorite gl-native PR in recent memory, 2) do you think their would be any performance benefits to letting CI skip builds for the node bindings as well?

Reply:
I didn’t add node as a skippable platform because:

- Those full jobs don’t take very long to finish.
- Node, Qt, and such aren't very actively developed and, when they are, tend to be in support of some core feature that would benefit from a full run.

Reply:
That said, if platform owners want to opt their own platforms/paths into this functionality, I’d encourage anyone to do so.